File: Assignment36_2.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : struprx
// Description   : Convert complete string into uppercase
// Input         : String
// Output        : Uppercase string
// Author        : Raj Samir Jadhav
// Date          : 07/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public String struprx(String str)   // Function returns modified string
    {
        String result = "";

        for(int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            if(ch >= 'a' && ch <= 'z')  // If lowercase, convert to uppercase
            {
                ch = (char)(ch - 32);
            }
            result = result + ch;
        }
        return result;
    }
}

class Assignment36_2
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        String output = pobj.struprx(input);

        System.out.println("Modified string is : " + output);

        sobj.close();
    }
}

File: Assignment36_2.c
#include <stdio.h>

////////////////////////////////////////////////////////////////////////////////
// Function Name : struprx
// Description   : Convert complete string into uppercase
// Input         : String
// Output        : Uppercase string
// Author        : Raj Samir Jadhav
// Date          : 07/12/2025
//////////////////////////////////////////////////////////////////////////////

void struprx(char *str)
{
    while(*str != '\0')                      // Traverse string
    {
        if((*str >= 'a') && (*str <= 'z'))   // Check for small letters
        {
            *str = *str - 32;                // Convert to uppercase (ASCII logic)
        }
        str++;
    }
}

int main()
{
    char arr[30];

    printf("Enter string : ");
    scanf("%[^'\n']s",arr);

    struprx(arr);                            // Function Call

    printf("Modified string is : %s\n",arr);

    return 0;
}
